src/components/fileviewer.js

```diff
@@ -111,3 +115,27 @@ const FileViewerMeta = ({ revision, path, appError }) => {
     </div>
   );
 };
+
+const CoverageMeta = ({ coverage }) => {
+  let [percentageCovered, totalCovered, totalUncovered] = [undefined, undefined, undefined];
+
+  if (coverage) {
+    /* TODO Currently showing coverage totals for one particular test:
+     * test-linux64-ccov/opt-web-platform-tests-1.
+     * In the future, test will be passed as a prop. */
+    const test = coverage.data[0].source.file;
+    this.percentageCovered=(test.percentage_covered * 100).toPrecision(4);
```

Comment:
the underscore (`_`) library may help here:

    let total_covered = _.union(coverage.data.forEach((d) => d.source.file.covered);
    let uncovered = _.union(coverage.data.forEach((d) => d.source.file.uncovered);
    let total_uncovered = _.difference(uncovered, total_covered);
    this.percentageCovered = total_covered.length / total_uncovered.length;

is did not test this code, or check for zeros, but I hope you get the idea.

Reply:
total_covered returns NaN because forEach doesn't return anything.  I have tried _.unionBy, using a combination of _.union and ._each, and using sets but no luck so far.  Can you please suggest me an alternative workaround?